env-data.sh

```diff
@@ -17,6 +17,34 @@ PGSTAT_TMP="/var/run/postgresql/"
 PG_PID="/var/run/postgresql/12-main.pid"
 
 
+# Read data from secrets into env variables.
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+function file_env {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+		exit 1
+	fi
+	local val="$def"
+	if [ "${!var:-}" ]; then
+		val="${!var}"
+	elif [ "${!fileVar:-}" ]; then
+		val="$(< "${!fileVar}")"
+	fi
+	export "$var"="$val"
+	unset "$fileVar"
+}
+
+file_env 'POSTGRES_PASS'
+file_env 'POSTGRES_USER'
+file_env 'POSTGRES_DBNAME'
+
```

Comment:
@gfkeith Should the secrets be for those three env variables only?. Could you update the documentation as well as adding a small section of how to utilize this in docker-compose in the readme

Reply:
There's no reason it should be limited to those - they're just the main ones I'd use. The docker library postgres (https://hub.docker.com/_/postgres/) has POSTGRES_INITDB_ARGS, POSTGRES_PASSWORD, POSTGRES_USER, and POSTGRES_DB. Yeah of course, I'll add a section on it.